multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/util/UnsupportedParameterFinder.java

```diff
@@ -0,0 +1,113 @@
+package org.cloudfoundry.multiapps.controller.process.util;
+
+import java.util.AbstractMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
+
+import jakarta.inject.Inject;
+import jakarta.inject.Named;
+import org.cloudfoundry.multiapps.mta.model.DeploymentDescriptor;
+import org.cloudfoundry.multiapps.mta.resolvers.CustomParameterContainer;
+import org.cloudfoundry.multiapps.mta.resolvers.Reference;
+import org.cloudfoundry.multiapps.mta.resolvers.ReferenceContainer;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.context.annotation.Scope;
+
+@Named("UnsupportedParameterFinder")
+@Scope(BeanDefinition.SCOPE_PROTOTYPE)
+public class UnsupportedParameterFinder {
+
+    @Inject
+    private SupportedParameterChecker supportedParameterChecker;
+
+    private static final String GLOBAL_PARAMETER = "global-parameter";
+
+    public Map<String, List<String>> findUnsupportedParameters(DeploymentDescriptor descriptor, List<ReferenceContainer> references) {
+        List<CustomParameterContainer> unknownParametersContainer = findUnsupportedParametersInDescriptor(descriptor);
+        return collectUnsupportedParametersByOwner(unknownParametersContainer, references);
+    }
+
+    private List<CustomParameterContainer> findUnsupportedParametersInDescriptor(DeploymentDescriptor descriptor) {
+        return supportedParameterChecker.getCustomParameters(descriptor);
+    }
+
+    private Map<String, List<String>> collectUnsupportedParametersByOwner(List<CustomParameterContainer> containers,
+                                                                          List<ReferenceContainer> references) {
+        return containers.stream()
+                         .map(container -> toFilteredEntry(container, references))
+                         .filter(this::hasUnsupportedParameters)
```

Comment:
this method is also maybe unnecessary and could be replaced with a !isEmpty()

Reply:
Done